oracle/src/events/processAMBCollectedSignatures/estimateGas.js

```diff
@@ -46,7 +46,7 @@ async function estimateGas({
 
     // check if all the signatures were made by validators
     for (let i = 0; i < v.length; i++) {
-      const address = web3.eth.accounts.recover(message, web3.utils.toHex(v[i]), r[i], s[i])
+      const address = web3.eth.accounts.recover(message, v[i], r[i], s[i])
```

Comment:
Previously, we had two functions `signatureToVRS` and `signatureToVRSAMB` in `oracle/src/utils/message.js`, I got rid of the first one and renamed the second one into `signatureToVRS`. Previously `v` component of a signatures was a number, now it is returned as a hex string.

However, I didn't notice that `recover` accepts only `0x`-prefixed hex strings. Fixed in last commits.

It is not necessary to patch existing instances, the previous version of the source code should work fine, since the logic itself was left unchanged.

Reply:
Thanks for clarification. But one thing is still unclear for me. You said that _`recover` accepts only 0x-prefixed hex strings._ At the same time _previously `v` component of a signatures was a number_. How did it work before then? Or probably you meant that previously `v` component of a signatures was a hex string and now it is a number that's why we need to convert it to a hex representation?

Reply:
I meant, that before `v` was a number, and we cast it to hex via `web3.utils.toHex(v[i])`. But, now all components `v`, `r`, `s` are just strings without a `0x` prefix, so it should be added explicitly.

Reply:
Ok. Got you. Thanks!